core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -273,15 +273,15 @@ public void repartition(final String topic, final int partitionsNumber) throws C
     public String createTopic(final NakadiTopicConfig nakadiTopicConfig) throws TopicCreationException {
 
         final KafkaTopicConfig kafkaTopicConfig = kafkaTopicConfigFactory.createKafkaTopicConfig(nakadiTopicConfig);
-        try (KafkaZkClient client = createZkClient()) {
-            final AdminZkClient adminZkClient = new AdminZkClient(client);
-            adminZkClient.createTopic(
-                    kafkaTopicConfig.getTopicName(),
-                    kafkaTopicConfig.getPartitionCount(),
-                    kafkaTopicConfig.getReplicaFactor(),
-                    kafkaTopicConfigFactory.createKafkaTopicLevelProperties(kafkaTopicConfig),
-                    kafkaTopicConfig.getRackAwareMode()
```

Comment:
this property is not set, how does that work ?

Reply:
If I understood it right (and I will test) this is also a broker property. So we don't need to specify it on the client. The new client does not have an algorithm for assigning partitions. It just tells the broker the number of partitions and the brokers does it. It does have an API for explicitly assigning partitions. If my assumption is wrong, I'll have to write some code that loads the brokers and the AZs and generates the assignment. But I think this is unnecessary. I couldn't find docs about it though so I have to test.

Reply:
@adyach I confirm that on the broker side there is logic for rack awareness and it uses it in case all brokers have rack's defined (which we do).

Kafka algorithm https://github.com/apache/kafka/blob/3e6dcb14dd9716f07d1ba96b1b06b78e55e92de1/core/src/main/scala/kafka/admin/AdminUtils.scala#L31-L123

Rack config on broker side https://github.com/zalando-nakadi/bubuku/blob/30086c0fb97d3e4d684ef2e34b135f1eaef5de83/bubuku/daemon.py#L52-L54